### PR TITLE
Add cmake workaround for hdf5 version detection when using hdf5 with subversion

### PR DESCRIPTION
**Pull request by Wentzell**

### CODE REVIEW

File: c++/h5/CMakeLists.txt [hunk 1]
```diff
@@ -27,13 +27,16 @@ install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} DESTINATION include FILES_MATCHING
...
 # Link against interface target and export
 target_link_libraries(h5_c PRIVATE hdf5::hdf5 hdf5::hdf5_hl)
-if(HDF5_VERSION VERSION_GREATER 1.10)
+if(HDF5_VERSION VERSION_GREATER 1.10 OR HDF5_SUBVERSION_ISSUE)
   target_compile_definitions(h5_c PRIVATE H5_USE_110_API)
 endif()
-if(HDF5_VERSION VERSION_GREATER_EQUAL 1.13)
+if(HDF5_VERSION VERSION_GREATER_EQUAL 1.13 OR HDF5_SUBVERSION_ISSUE)
   target_compile_definitions(h5_c PRIVATE H5_VER_GE_113)
 endif()
```
Comment (hmenke) [inline c++/h5/CMakeLists.txt]: I wonder why we need the version detection at all. The names of the macros do not suggest that these macros would have any effect on older versions at all, so I think the most pragmatic solution would be to enable these unconditionally.

Comment (hmenke) [reply c++/h5/CMakeLists.txt]: Another option would be to negate the logic, i.e.
```cmake
if(NOT HDF5_VERSION VERSION_LESS 1.10)
  target_compile_definitions(h5_c PRIVATE H5_USE_110_API)
endif()
if(NOT HDF5_VERSION VERSION_LESS 1.13)
  target_compile_definitions(h5_c PRIVATE H5_VER_GE_113)
endif()
```
under the assumption that the HDF5 version detection is not broken for these older versions.

**Comment by Wentzell** [comment]
Thank you @hmenke for the suggestions, I have updated the PR accordingly.

**Comment by Wentzell** [comment]
Note that the `1.13` version check is necessary and can't be dropped

File: c++/h5/CMakeLists.txt [hunk 2]
```diff
@@ -30,10 +30,8 @@ find_package(HDF5 REQUIRED C HL)
 
 # Link against interface target and export
 target_link_libraries(h5_c PRIVATE hdf5::hdf5 hdf5::hdf5_hl)
-if(HDF5_VERSION VERSION_GREATER 1.10)
-  target_compile_definitions(h5_c PRIVATE H5_USE_110_API)
-endif()
-if(HDF5_VERSION VERSION_GREATER_EQUAL 1.13)
+target_compile_definitions(h5_c PRIVATE H5_USE_110_API)
+if(NOT HDF5_VERSION VERSION_LESS 1.13)
```
Comment (hmenke) [inline c++/h5/CMakeLists.txt]: I was too optimistic. This doesn't work when `HDF5_VERSION` is missing altogether.

**Comment by Wentzell** [comment]
Ok. Can I ask which hdf5 versions are affected by this issue?

**Comment by hmenke** [comment]
This started with 1.14.1. Before that the `H5_VERS_SUBRELEASE` suffix was only for release candidates, but with version `1.14.1-2` they started tagging full releases with it. https://github.com/HDFGroup/hdf5/tags

**Review by hmenke — APPROVED** [review]
Smart move ;)

**Pull request merged**